python/lsst/summit/utils/utils.py

```diff
@@ -1419,3 +1425,221 @@ def getSunAngle(time: Time | None = None) -> float:
+        self.residual_threshold = residual_threshold
+        self.random_state = random_state
+
+        if self.residual_threshold is None:
+            # Set a default residual threshold based on the data
+            if np.isnan(x).all() or np.isnan(y).all():
+                raise ValueError("Cannot fit model: all input data is NaN.")
+            self.residual_threshold = 1.5 * mad_std(y)
+
+        self.fit(x, y)
+
+    def fit(self, x, y):
+        # Ensure x and y are numpy arrays
+        x = np.asarray(x)
+        y = np.asarray(y)
+
+        # check for NaNs
+        is_nan = np.isnan(x) | np.isnan(y)
+        if np.all(is_nan):
+            raise ValueError("All input data is NaN. Cannot fit a model.")
+
+        X = np.asarray(x[~is_nan]).reshape(-1, 1)
+        yfit = np.asarray(y[~is_nan])
+
+        self.model = RANSACRegressor(
+            estimator=LinearRegression(),
+            min_samples=self.min_samples,
+            residual_threshold=self.residual_threshold,
+            random_state=self.random_state,
+        )
+        self.model.fit(X, yfit)
+        self.slope = self.model.estimator_.coef_[0]
+        self.intercept = self.model.estimator_.intercept_
+
+        # Flatten x to ensure it is 1D for statsmodels
+        self.x = x.flatten()
+        self.y = np.asarray(y)
+
+        # Use only inliers for statistics
+        self.outlier_mask = np.full(self.x.shape, False, dtype=bool)
+        self.outlier_mask[is_nan] = True  # Keep NaNs as outliers
+        # Mark outliers as True, inliers as False
+        self.outlier_mask[~is_nan] = ~self.model.inlier_mask_
+
+        x_inlier = self.x[~self.outlier_mask]
+        y_inlier = self.y[~self.outlier_mask]
+
+        # OLS with statsmodels for CI and p-value
+        X_design = sm.add_constant(x_inlier)
+        ols_model = sm.OLS(y_inlier, X_design).fit()
+
+        # statsmodels reports [intercept, slope]
+        ci = ols_model.conf_int(alpha=0.32)  # 68% CI
+        self.ci_intercept = ci[0]
+        self.ci_slope = ci[1]
+
+        self.slope_pvalue = ols_model.pvalues[1]
+        self.slope_stderr = ols_model.bse[1]
+        self.slope_tvalue = ols_model.tvalues[1]
+        self.scatter = np.std(ols_model.resid)
+        self.ols_model = ols_model  # Expose full model if needed
+
+    def report_best_values(self):
+        return RobustFitResult(
+            slope=self.slope,
+            slope_pvalue=self.slope_pvalue,
+            slope_stderr=self.slope_stderr,
+            slope_tvalue=self.slope_tvalue,
+            intercept=self.intercept,
+            intercept_pvalue=self.ols_model.pvalues[0],
+            intercept_stderr=self.ols_model.bse[0],
+            intercept_tvalue=self.ols_model.tvalues[0],
+            scatter=self.scatter,
+        )
+
+    def plot_best_fit(
+        self,
+        ax=None,
+        label=None,
+        color=None,
+        alpha_band=0.2,
+        lw=2,
+        nbins=5,
+        is_scatter=False,
```

Comment:
please mypy at least the bits which aren't matplotlib objects just to help people out and provide a little safety